cpu/esp_common/periph/flash.c

```diff
@@ -460,7 +462,7 @@ const esp_partition_t* esp_partition_find_first(esp_partition_type_t type,
                 part->address = info->pos.offset;
                 part->size = info->pos.size;
                 part->encrypted = info->flags & PART_FLAG_ENCRYPTED;
-                strncpy(part->label, (const char*)info->label, sizeof(info->label));
+                strncpy(part->label, (const char*)info->label, ARRAY_SIZE((info->label)));
```

Comment:
Why `ARRAY_SIZE` for `strncpy`?

Reply:
Otherwise, you get a compilation error:
```
cpu/esp_common/periph/flash.c:465:70: error: argument to 'sizeof' in 'strncpy' call is the same expression as the source; did you mean to use the size of the destination? [-Werror=sizeof-pointer-memaccess]
                 strncpy(part->label, (const char*)info->label, sizeof(info->label));
```

Reply:
`info->label` fact it is an character array.

Reply:
Ah but the warning is correct. You are copying into `part->label`, so you should use `sizeof(part->label)`.

Reply:
Oh yeah, of course.